src/main/java/fruit/shop/service/transaction/PurchaseTransactionHandler.java

```diff
@@ -0,0 +1,19 @@
+package fruit.shop.service.transaction;
+
+import fruit.shop.db.Storage;
+import fruit.shop.model.FruitTransaction;
+
+public class PurchaseTransactionHandler implements TransactionHandler {
+    private final QuantityValidator quantityValidator = new QuantityValidator();
+
+    @Override
+    public void execute(FruitTransaction transaction) {
+        Integer stock = Storage.get(transaction.getFruit());
+        Integer transactionQuantity = transaction.getQuantity();
+        if (quantityValidator.validate(stock, transactionQuantity)) {
+            Storage.put(transaction.getFruit(), stock - transactionQuantity);
+            return;
+        }
+        throw new RuntimeException("Shop doesn't have enough stock of: " + transaction.getFruit());
```

Comment:
This case is pretty trivial, but for more complex code you might want to reverse the order:
1. Go through negative case scenarios and end execution when it occurrs (`if (!quantityValidator.validate()...)`)
2. Process the positive scenario (`Storage.put(...)`)

you lose the obscure `return` call.